Add Benchmark glossary (#8190)
opensearch-project/documentation-website: _benchmark/glossary.md

---
layout: default
title: Glossary
nav_order: 10
---

# OpenSearch Benchmark glossary

The following terms are commonly used in OpenSearch Benchmark:

> - **Corpora**: A collection of documents.

Reviewer: Line 12: "equivalent to" => "equal to"? ("equal" meaning "same in all respects," "equivalent" meaning "similar but not identical")

> - **Latency**: Based on the `target-throughput` set by the user, the total amount of time that the request waits before receiving the response, in addition to any other delays that occur before the request is sent.

Author: ```suggestion
- **Latency**: If `target-throughput` is disabled (has no value or a value of `0)`, latency is equivalent to service time. If `target-throughput` is enabled (has a value of 1 or greater), latency is the service time plus the time the request waits in the queue before being sent.
```

> - **Latency**: If `target-throughput` is disabled (has no value or a value of `0)`, latency is equivalent to service time. If `target-throughput` is enabled (has a value of 1 or greater), latency is the service time plus the time the request waits in the queue before being sent.

Reviewer: ```suggestion
- **Latency**: If `target-throughput` is disabled (has no value or a value of `0)`, then latency is equivalent to service time. If `target-throughput` is enabled (has a value of 1 or greater), then latency is equal to the service time plus the amount of time the request waits in the queue before being sent.
```

> - **Latency**: If `target-throughput` is disabled (has no value or a value of `0)`, then latency is equivalent to service time. If `target-throughput` is enabled (has a value of 1 or greater), then latency is equal to the service time plus the amount of time the request waits in the queue before being sent.

Author: ```suggestion
- **Latency**: If `target-throughput` is disabled (has no value or a value of `0)`, then latency is equal to service time. If `target-throughput` is enabled (has a value of 1 or greater), then latency is equal to the service time plus the amount of time the request waits in the queue before being sent.
```

> - **Metric keys**: The metrics that OpenSearch Benchmark stores, based on the configuration in the [metrics record]({{site.url}}{{site.baseurl}}/benchmark/metrics/metric-records/).

Reviewer: ```suggestion
- **Metric keys**: The metrics stored by OpenSearch Benchmark, based on the configuration in the [metrics record]({{site.url}}{{site.baseurl}}/benchmark/metrics/metric-records/).
```

> - **Operations**: In workloads, a list of API requests performed by a workload.

Reviewer 2: Should we say "API operations for consistency?

> - **Operations**: In workloads, a list of API requests performed by a workload.

Author: ```suggestion
- **Operations**: In workloads, a list of API operations performed by a workload.
```

> - **Pipeline**: A series of steps occurring before and after a workload is run that determines benchmark results.

Reviewer: ```suggestion
- **Pipeline**: A series of steps occurring both before and after running a workload that determines benchmark results.
```

> - **Schedule**: In workloads, a list of operations in a specific order.

Reviewer 2: So, the difference between this and `operations` is that `operations` is not necessarily in order but `schedule` is? Also, is the order time-based, and if so, is it execution time or the time the request was put in the queue or sent?

Author: Not exactly. A schedule is simply a list of one or more operations performed in the order they appear at the time the workload is run. The order the operations in a schedule isn't time-based. 

I'll adjust the definition accordingly.

> - **Schedule**: In workloads, a list of operations in a specific order.

Author: ```suggestion
- **Schedule**: A list of two or more operations performed in the order they appear when a workload is run.
```

> - **Service time**: The amount of time that it takes for `opensearch-py` to send a request and receive a response from the OpenSearch cluster. It includes the amount of time that it takes for the server to process a request and also _includes_ network latency, load balancer overhead, and deserialization/serialization.

Reviewer 3: Nit: for `opensearch-py`, the primary client of OpenSearch Benchmark,

> - **Service time**: The amount of time that it takes for `opensearch-py`, the primary client for OpenSearch Benchmark, to send a request and receive a response from the OpenSearch cluster. It includes the amount of time that it takes for the server to process a request and also _includes_ network latency, load balancer overhead, and deserialization/serialization.

Reviewer 2: ```suggestion
- **Service time**: The amount of time that it takes for `opensearch-py`, the primary client for OpenSearch Benchmark, to send a request and receive a response from the OpenSearch cluster. It includes the amount of time that it takes for the server to process a request and also network latency, load balancer overhead, and deserialization/serialization.
```

> - **Service time**: The amount of time that it takes for `opensearch-py`, the primary client for OpenSearch Benchmark, to send a request and receive a response from the OpenSearch cluster. It includes the amount of time that it takes for the server to process a request and also network latency, load balancer overhead, and deserialization/serialization.

Reviewer: ```suggestion
- **Service time**: The amount of time taken for `opensearch-py`, the primary client for OpenSearch Benchmark, to send a request and receive a response from the OpenSearch cluster. It includes the amount of time taken for the server to process a request as well as for network latency, load balancer overhead, and deserialization/serialization.
```

> - **Summary report**: A report output at the end a test based on the metric keys defined in the workload.

Reviewer: ```suggestion
- **Summary report**: A report generated at the end of a test based on the metric keys defined in the workload.
```

- **Test**: A single invocation of the OpenSearch Benchmark binary.
- **Throughput**: The number of operations completed in a given period of time.

> - **Workload**: A collection of one or more benchmarking scenarios that use a specific document corpus to perform a benchmark against your cluster. The document corpus contains any indexes, data files, and operations invoked when the workload runs.

Reviewer 2: Is a scenario the same as a test?

> - **Workload**: A collection of one or more benchmarking scenarios that use a specific document corpus to perform a benchmark against your cluster. The document corpus contains any indexes, data files, and operations invoked when the workload runs.

Author: ```suggestion
- **Workload**: A collection of one or more benchmarking tests that use a specific document corpus to perform a benchmark against your cluster. The document corpus contains any indexes, data files, and operations invoked when the workload runs.
```

> - **Workload**: A collection of one or more benchmarking tests that use a specific document corpus to perform a benchmark against your cluster. The document corpus contains any indexes, data files, and operations invoked when the workload runs.

Reviewer: ```suggestion
- **Workload**: A collection of one or more benchmarking tests that use a specific document corpus to perform a benchmark against a cluster. The document corpus contains any indexes, data files, or operations invoked when the workload runs.
```
